series/series.go

```diff
@@ -437,37 +519,59 @@ func (s Series) Compare(comparator Comparator, comparando interface{}) Series {
 		return Bools(bools)
 	}
 
-	// Single element comparison
-	if comp.Len() == 1 {
+	// Multiple element comparison of single value types
+	if s.Type() == String ||
+		s.Type() == Int ||
+		s.Type() == Float ||
+		s.Type() == Bool {
+		// Single element comparison
+		if comp.Len() == 1 {
+			for i := 0; i < s.Len(); i++ {
+				e := s.elements.Elem(i)
+				c, err := compareElements(e, comp.elements.Elem(0), comparator)
+				if err != nil {
+					s = s.Empty()
+					s.Err = err
+					return s
+				}
+				bools[i] = c
+			}
+			return Bools(bools)
+		}
+
+		// Multiple element comparison of single value types
+		if s.Len() != comp.Len() {
+			s := s.Empty()
+			s.Err = fmt.Errorf("can't compare: length mismatch")
+			return s
+		}
 		for i := 0; i < s.Len(); i++ {
 			e := s.elements.Elem(i)
-			c, err := compareElements(e, comp.elements.Elem(0), comparator)
+			c, err := compareElements(e, comp.elements.Elem(i), comparator)
 			if err != nil {
 				s = s.Empty()
 				s.Err = err
 				return s
 			}
 			bools[i] = c
 		}
+	} else {
+		// Single element comparison
+		if comp.Len() == 1 {
```

Comment:
Can you clarify what's the behavior of list type comparison? It'll be great if you add those informations in the method's documentation too.